src/page/HomePage.tsx

```diff
@@ -1,123 +1,74 @@
-  const selectedTodo = useSelector((state: RootState) =>
-    todoId ? selectTodoById(todoId)(state) : null
-  );
+  const { getProjectData } = useProjectPage();
+  const { isCreateProject } = useForm();
+  const { isLoading, setIsLoading, projectClick, trashOnClick, addProjectClick } = useHomePage();
+  const projects = useSelector((state: RootState) => state.projectSlice.projects);
 
   useEffect(() => {
     if (!user) {
-      navigateToLoginPage('/login');
+      navigateToLoginPage(MyRoutes.LOGIN_ROUTE);
     }
   }, [user]);
 
   useEffect(() => {
     getProjectData();
+    setIsLoading(false);
   }, []);
 
-  useEffect(() => {
-    getDataHandler();
-  }, [idActiveProject]);
-
-  useEffect(() => {
-    if (setTextareaValue) {
-      setTextareaValue('');
-    }
-    if (setTitleValue) {
-      setTitleValue('');
-    }
-    if (setSelectValue && selectedTodo) {
-      setSelectValue(selectedTodo.importance);
-    } else {
-      setSelectValue!('Low');
-    }
-  }, [createTodo, isVisible]);
-
   return (
-    <div className="overscroll-x-none overflow-x-hidden">
-      <Header />
-      <div className="flex py-2  items-center  overflow-hidden w-screen h-screen relative ">
-        <Sidebar />
+    <div className="overscroll-x-none overflow-x-hidden bg-slate-100">
+      <Header value="Home" />
+      <HomePageWrapper>
         {isLoading ? (
           <Loader />
-        ) : idActiveProject ? (
-          <Content
-            blocks={blocks}
-            dragDropHandler={dragDropHandler}
-            dragEndHandler={dragEndHandler}
-            dragStartHandler={dragStartHandler}
-            dragOverHandler={dragOverHandler}
-            dragEnterHandler={dragEnterHandler}
-            dragLeaveHandler={dragLeaveHandler}
-          />
         ) : (
-          <div>none</div>
-        )}
-
-        {createTodo && (
-          <Popup
-            titleInputChangeHandler={titleInputChangeHandler}
-            textareaChangeHandler={textareaChangeHandler}
-            selectChangeHandler={selectChangeHandler}
-            setCreateTodo={setCreateTodo}
-            createTodo={createTodo}
-            selectValue={selectValue}
-            submitButtonHandler={submitButtonHandler}
-          />
-        )}
-        {isVisible && (
-          <Popup
-            titleInputChangeHandler={titleInputChangeHandler}
-            textareaChangeHandler={textareaChangeHandler}
-            selectChangeHandler={selectChangeHandler}
-            selectValue={selectValue}
-            deleteButtonHandler={deleteButtonHandler}
-            updateButtonHandler={updateButtonHandler}
-            selectedTodo={selectedTodo}
-          />
+          <>
+            <h1 className="mb-2 text-2xl">Your Projects:</h1>
+            <HomeProjectWrapper>
+              {projects.map((item, index) => {
+                return (
+                  <ProjectBlockHomeWrapper key={index} onClick={() => projectClick(item)}>
```

Comment:
let's create seperate component for mapping and create handler for onClick
same as above about index